src/meta/meta_backup_service.cpp

```diff
@@ -1569,5 +1848,40 @@ std::string backup_service::get_backup_path(const std::string &policy_name, int6
     ss << _policy_meta_root << "/" << policy_name << "/" << backup_id;
     return ss.str();
 }
+
+void backup_service::start_backup_apps(start_backup_apps_rpc rpc)
+{
+    const start_backup_apps_request &request = rpc.request();
+    start_backup_apps_response &response = rpc.response();
+
+    std::shared_ptr<backup_engine> engine = std::make_shared<backup_engine>(this);
```

Comment:
I think unique_ptr is better here

Reply:
The pointer would be used to query backup status, so it stored as a shared_ptr in the map `_backup_states`.